### DISCUSSION TITLE
umi4下react-router和react-router-dom可否升级到6.7.0以上

### DISCUSSION THREAD

**Discussion by wtchaos** [umijs/umi#11011] [Q&A]
react-router在v6之后移除了Prompt的功能，之后又在6.7.0版本重新加入了此功能，而目前umi讲react-router版本固定在了6.3.0，可否将版本升级到6.7.0以上来支持Prompt功能

**Comment by fz6m** [comment]
加上的是 data browser router 的 Prompt ，目前 umi 4 内部用的是 browser router ，升级了 react router 也用不了的。

暂时解决方案：https://github.com/umijs/umi/discussions/10237#discussioncomment-4625915

**Reply by wtchaos** [reply]
尝试了这个方案的代码，会在点击浏览器后退时，地址栏会先变成上一页的地址，在瞬间变回来

**Reply by fz6m** [reply]
有什么缺点吗

**Reply by wtchaos** [reply]
有些业务逻辑需要监听location的变化，地址栏瞬间变化会使监听回调执行，但是后退被阻止的话这次回调不应该被执行

**Reply by fz6m** [reply]
react router `6.7.0` 新增的 `unstable_usePrompt` 也是不稳定的 api ，也会发生你说的闪路径问题。

根本原因是 react router v6 的设计就是瞬时路径切换，这也使得路由动画做不了了，升级 v6 最新版本没法解决你说的问题。

**Reply by wtchaos** [reply]
之前使用umi3好像没有发现这样的问题，我目前使用history.pushState()暂时实现了我这边的需求

**Reply by fz6m** [reply]
umi 3 不是 react router v6 ，你说的这个问题不是 umi 的问题，react router 的迭代设计就是如此。

**Reply by liujie3612** [reply]
umi 4 有什么办法使用router5么

**Comment by undefined996** [comment]
umi4下有什么办法实现history.listen么? umi3的时候useLocation返回的值会带有listen

**Reply by fz6m** [reply]
直接导入 history 就可以调用 `history.listen` 了。
